glusterd2/volume/volume-utils.go

```diff
@@ -309,26 +310,25 @@ func CleanBricks(volinfo *Volinfo) error {
 			continue
 		}
 
-		// Remove Thin Pool if LV count is zero, Thinpool will
-		// have more LVs in case of snapshots and clones
-		numLvs, err := lvmutils.NumberOfLvs(vgname, tpname)
+		err = lvmutils.DeactivateLV(vgname, tpname)
 		if err != nil {
 			log.WithError(err).WithFields(log.Fields{
 				"vg-name": vgname,
 				"tp-name": tpname,
-			}).Error("failed to get number of lvs")
+			}).Error("thinpool deactivate failed")
 			return err
 		}
 
-		if numLvs == 0 {
-			err = lvmutils.RemoveLV(vgname, tpname)
-			if err != nil {
-				log.WithError(err).WithFields(log.Fields{
-					"vg-name": vgname,
-					"tp-name": tpname,
-				}).Error("thinpool remove failed")
-				return err
-			}
+		err = lvmutils.RemoveLV(vgname, tpname, false)
+		// Do not remove Thinpool if the dependent Lvs exists
+		// Ignore the lvremove command failure if the reason is
+		// dependent Lvs exists
+		if err != nil && !lvmutils.IsDependentLvsError(err) && !lvmutils.IsLvNotFoundError(err) {
```

Comment:
Why is LV not found error being ignored?

Reply:
for repeat-ability. Previously we observed the volume delete failures due to previous transaction failures.(If LVs deleted but crashed before removing from store)